rust/generate_message/src/helpers.rs

```diff
@@ -70,10 +80,14 @@ pub fn network_specs_from_entry(
 ///
 /// If the [`NetworkSpecsKey`] and associated [`NetworkSpecsToSend`] are not
 /// found in the [`SPECSTREEPREP`], the result is `Ok(None)`.
-pub fn try_get_network_specs_to_send(
+pub fn try_get_network_specs_to_send<P>(
     network_specs_key: &NetworkSpecsKey,
-) -> Result<Option<NetworkSpecsToSend>> {
-    let database = open_db(HOT_DB_PATH.to_str().unwrap())?;
+    db_path: P,
+) -> Result<Option<NetworkSpecsToSend>>
+where
+    P: AsRef<Path>,
+{
+    let database = open_db(db_path.as_ref().to_str().unwrap())?;
```

Comment:
So there is no any reason left for `open_db` and others to take `&str` as a param now, it can also be generic over `AsRef<Path>` just like `sled::open` is.

Reply:
Totally agree, but that would trigger changes in many other places and it might be hard to review this PR.

Reply:
Right. My concern was that it will trigger changes in many places, and it'll harder to review this PR.